### PR TITLE
Implement sparse option in create_index

### PR DESCRIPTION
**Pull request by DhirajS**
Implement sparse option in create_index

### CODE REVIEW

**Comment by DhirajS** [comment]
@hiteshghia-splunk @jsakai @joelatsplunk  Please take a look this PR to support sparse option in create_index()

File: mongomock/collection.py [hunk 1]
```diff
@@ -293,9 +294,16 @@ def _insert(self, data):
...
-            answer = self.find(find_kwargs)
-            if answer.count() > 0:
-                raise DuplicateKeyError("Duplicate Key Error", 11000)
+
+            for key in find_kwargs.keys():
+                if key in self._sparses:
+                    if find_kwargs[key] == None:
+                        del find_kwargs[key]
+
+            if find_kwargs != {}:
```
Comment (joelatsplunk) [inline mongomock/collection.py]: should be:
if not find_kwargs:

File: mongomock/collection.py [hunk 2]
```diff
@@ -293,9 +294,16 @@ def _insert(self, data):
...
+                answer = self.find(find_kwargs)
+                if answer.count() > 0:
+                    raise DuplicateKeyError("Duplicate Key Error", 11000)
```
Comment (joelatsplunk) [inline mongomock/collection.py]: single quotes!

File: mongomock/collection.py [hunk 3]
```diff
@@ -293,9 +294,16 @@ def _insert(self, data):
...
-                raise DuplicateKeyError("Duplicate Key Error", 11000)
+
+            for key in find_kwargs.keys():
+                if key in self._sparses:
+                    if find_kwargs[key] == None:
+                        del find_kwargs[key]
+
+            if find_kwargs != {}:
+                answer = self.find(find_kwargs)
+                if answer.count() > 0:
```
Comment (joelatsplunk) [inline mongomock/collection.py]: > 0 is superfluous

**Comment by joelatsplunk** [comment]
⚡️ but a couple nits

**Comment by joelatsplunk** [comment]
⚡️

File: mongomock/collection.py (hunk 3 above)
Comment (DhirajS) [inline mongomock/collection.py]: @joelatsplunk I like to be defensive with checks like this one. If there is some internal error in find that causes it to return a negative integer for count() when answer is in fact empty, I do not want this to fail.

**Pull request merged**